llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -887,6 +887,20 @@ static void simplifyRecipe(VPRecipeBase &R, VPTypeAnalysis &TypeInfo) {
     Blend->replaceAllUsesWith(NewBlend);
     Blend->eraseFromParent();
     recursivelyDeleteDeadRecipes(DeadMask);
+
+    /// Simplify BLEND %a, %b, Not(%mask) -> BLEND %b, %a, %mask.
+    VPValue *NewMask;
+    if (NewBlend->getNumOperands() == 3 &&
+        match(NewBlend->getMask(1), m_Not(m_VPValue(NewMask)))) {
+      VPValue *Inc0 = NewBlend->getIncomingValue(0);
+      VPValue *Inc1 = NewBlend->getIncomingValue(1);
+      VPValue *OldMask = NewBlend->getOperand(2);
+      NewBlend->setOperand(0, Inc1);
+      NewBlend->setOperand(1, Inc0);
+      NewBlend->setOperand(2, NewMask);
```

Comment:
For consistency, this should all use getMask(1)/setMask(1), getIncomingValue(0,1)/setIncomingValue(0,1), with setters to be added, or all use getOperand(0,1,2)/setOperand(0,1,2) as in the following.
BTW, VPBlendRecipe could be a VPInstruction, or a derivative of it - which provides additional methods (but not fields).
```suggestion
        match(NewBlend->getOperand(2), m_Not(m_VPValue(NewMask)))) {
      VPValue *IncomingValue0 = NewBlend->getOperand(0);
      VPValue *IncomingValue1 = NewBlend->getOperand(1);
      VPValue *OldMask = NewBlend->getOperand(2);
      NewBlend->setOperand(0, IncomingValue1);
      NewBlend->setOperand(1, IncomingValue0);
      NewBlend->setOperand(2, NewMask);
```

Reply:
Updated to use getOperand consistently for now, thanks!